pkg/kubelet/stats/cadvisor_stats_provider.go

```diff
@@ -317,6 +317,8 @@ func removeTerminatedContainerInfo(containerInfo map[string]cadvisorapiv2.Contai
 			podRef:        buildPodRef(cinfo.Spec.Labels),
 			containerName: kubetypes.GetContainerName(cinfo.Spec.Labels),
 		}
+		// Clear the UID since the container can be created in a new namespace.
+		cinfoID.podRef.UID = ""
```

Comment:
so buildPodRef will have set the podRef.UID  , we then strip it, wont every result in infos in ListPodCPUAndMemoryStats have an empty UID?

Reply:
which if i look at this code, makes me think the pod ref in pod stats will always have an empty uid.

Reply:
this is ok since `removeTerminatedContainerInfo` only sets up the slice to iterate around. This line iterates around the active containers, and initialized `podToStats` which contains the correct UID to the container namespace. https://github.com/kubernetes/kubernetes/blob/c677061be2cc4aa82168633fb7a19a5ce6dea945/pkg/kubelet/stats/cadvisor_stats_provider.go#L93

Reply:
ah ok, i missed that.

Reply:
ah missed that!  thanks for clarifiying.